scripts/create-agent.ts

```diff
@@ -367,6 +369,49 @@ AWS_DEFAULT_REGION=us-east-1`;
     }
   }
 
+  private async waitForInstanceHealthy(instanceId: string): Promise<void> {
+    const isCI = process.env.GITHUB_ACTIONS === 'true' || 
+                 process.env.CI === 'true' || 
+                 process.env.GITHUB_TOKEN;
+    
+    if (isCI) {
+      console.log("⚠️  Skipping health check in CI environment due to AWS account limitations");
+      console.log("✅ Agent setup completed (health check skipped in CI)");
+      return;
+    }
+    
+    console.log("Waiting for agent to be healthy...");
+    
+    const maxAttempts = 40;
+    let attempts = 0;
+    
+    while (attempts < maxAttempts) {
+      try {
+        const healthResult = await checkInstanceHealth(instanceId, this.config.region);
+        
+        if (healthResult.status === "healthy") {
+          console.log("✅ Agent is healthy and running");
+          return;
+        }
+        
+        attempts++;
+        const waitTime = attempts < 10 ? 15 : 30;
+        console.log(`Agent not healthy yet (${healthResult.status}${healthResult.error ? `: ${healthResult.error}` : ''}), waiting... (${attempts}/${maxAttempts})`);
+        await new Promise(resolve => setTimeout(resolve, waitTime * 1000));
+        
+      } catch (error) {
+        attempts++;
+        if (attempts >= maxAttempts) {
+          throw new Error(`Agent failed to become healthy within timeout: ${error instanceof Error ? error.message : String(error)}`);
```

Comment:
abstract this `error instanceof Error ? error.message : String(error)` piece out to a constant